packages/web/src/components/table-handle/dnd.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  createComputed,
+  useEffect,
+  type ConnectableElement,
+  type ReadonlySignal,
+} from '@aria-ui/core'
+import {
+  computePosition,
+  offset,
+} from '@floating-ui/dom'
+import type { Editor } from '@prosekit/core'
+import type { EditorView } from '@prosekit/pm/view'
+
+import {
+  tableHandleDndContext,
+  tableHandleRootContext,
+} from './context'
+
+export type OnInitParams = {
+  host: ConnectableElement
+  direction: 'row' | 'col'
+  dragging: boolean
+  draggingIndex: number
+  table: HTMLTableElement
+  cell: HTMLTableCellElement
+}
+
+export function useInitDndPosition(
+  host: ConnectableElement,
+  editor: ReadonlySignal<Editor | null>,
+  onInit: (params: OnInitParams) => void,
+): void {
+  const dndContext = tableHandleDndContext.consume(host)
+  const rootContext = tableHandleRootContext.consume(host)
+
+  const draggingSignal = createComputed(() => {
+    const context = dndContext.get()
+    return context.dragging
+  })
+
+  const directionSignal = createComputed(() => {
+    const context = dndContext.get()
+    return context.direction
+  })
+
+  const draggingIndexSignal = createComputed(() => {
+    const context = dndContext.get()
+    return context.draggingIndex
+  })
+
+  useEffect(host, () => {
+    const editorInstance = editor.get()
+    if (!editorInstance) return
+
+    const dragging = draggingSignal.get()
+    const direction = directionSignal.get()
+
+    host.dataset.direction = direction
+    host.dataset.dragging = dragging.toString()
+
+    const draggingIndex = draggingIndexSignal.get()
+    const { view } = editorInstance
+
+    const relatedDOMs = getDndRelatedDOMs(view, rootContext.peek()?.cellPos, draggingIndex, direction)
+    if (!relatedDOMs) return
+    const { table, cell } = relatedDOMs
+
+    onInit({
+      host,
+      direction,
+      dragging,
+      draggingIndex,
+      table,
+      cell,
+    })
+
+    if (!dragging) return
+
+    computePosition(cell, host, {
+      placement: direction === 'row' ? 'right' : 'bottom',
+      middleware: [offset(({ rects }) => {
+        if (direction === 'col') {
+          return -rects.reference.height
+        }
+
+        return -rects.reference.width
+      })],
+    })
+      .then(({ x, y }) => {
+        Object.assign(host.style, {
+          left: `${x}px`,
+          top: `${y}px`,
+        })
+      })
+      .catch((error) => {
+        console.error(error)
+      })
+  })
+}
+
+function getTableDOMByPos(view: EditorView, pos: number): HTMLTableElement | undefined {
+  const dom = view.domAtPos(pos).node
+  if (!dom) return
+  const element = dom instanceof HTMLElement ? dom : dom.parentElement
```

Comment:
`instanceof` in JavaScript world isn't reliable (and that's why we need something like [Error.isError()](https://github.com/tc39/proposal-is-error)). 

You can use the util before for this purpose. 

```js
import { isHTMLElement } from '@ocavue/utils'
```

Reply:
Fixed by https://github.com/prosekit/prosekit/pull/1109/commits/ed32c389b260f80606a576194a3283e2ad1243f8